### ISSUE TITLE
[java] DoNotUseThreads: False positive on legitimate java.lang.Thread.onSpinWait() call

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6520]
**Affects PMD Version:** 7.22.0

**Rule:**[DoNotUseThreads](https://pmd.github.io/pmd/pmd_rules_java_multithreading.html#donotusethreads)

**Description:**
The DoNotUseThreads rule is intended to prevent applications from creating or managing their own threads/ExecutorServices (as required by the J2EE specification).  
However, it **incorrectly triggers** on the static method call `java.lang.Thread.onSpinWait()`, which is **not** thread creation or management — it is a performance hint (intrinsic) introduced in Java 9 to help the JIT optimize spin-wait loops. This call appears in official OpenJDK jtreg tests and many legitimate performance-critical codes.

**Code Sample demonstrating the issue:**

```java
public class a{
    public void foo(){
        java.lang.Thread.onSpinWait();
    }
}
```

**Expected outcome:**

PMD reports a violation at line 3 for DoNotUseThreads ("To be compliant to J2EE, a webapp should not use any thread."), but that's wrong. That's a false positive.

**Running PMD through:** *CLI*
